### PR TITLE
rename test/e2e{,_node} api imports to groupversion

### PR DESCRIPTION
**Pull request by spiffxp**
**What type of PR is this?**
/kind cleanup
/area conformance

**What this PR does / why we need it**:
To help with conformance test review, let's explicitly include the version in all k8s api imports used in test/e2e and test/e2e_node. Further, for every api group except core, explicitly include the group name. If it's a component api, put that in the name as well.

Examples:
- groupversion "k8s.io/api/group/version"
- verson "k8s.io/api/core/version"
- kubeletgroupversion "k8s.io/kubernetes/pkg/kubelet/apis/group/version"

The exception for core is just because it'd involve changing way too
many files, most everything already uses "v1" implicitly or explicitly.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
I did a survey of our conventions on importing api aliases here https://gist.github.com/spiffxp/761ec887cde75e3d7aa47e0546c9cc97 and found they're complete entropy across kubernetes/kubernetes, but close to this proposed convention in test/e2e

I used https://gist.github.com/spiffxp/e278d28755a3f64899aaed8f73c78ecf on top of https://github.com/kubernetes/kubernetes/pull/78836 to generate these commits

This was motivated by https://github.com/kubernetes/kubernetes/pull/78614

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by draveness** [comment]
Maybe we could also import `"k8s.io/api/core/v1"` as `corev1`, but it could be a lot of work...

> Standard style is to call this package corev1
> 
> https://github.com/kubernetes/kubernetes/search?utf8=%E2%9C%93&q=corev1&type=
>
> With modules, many editors explicitly adding the v1 alias to the front to make it more obvious to people that this is a v1 package and not a v1 version of the core package, but most places this is imported we call it corev1
> https://github.com/kubernetes/kubernetes/pull/77336#discussion_r281783414

cc/ @johnSchnake

**Comment by spiffxp** [comment]
yeah empirically `v1` wins here, I would be fine with `corev1` but it'd be a large change, and that's about the point at which I would want to think about enforcing this via tooling (which may be more effort than it's worth, I dunno.. `goimports` didn't seem to lend itself to customization when I glanced)

our style all over k/k is all over the map, FWIW, it's just close enough to consistent in test that I'm nudging it forward

**Comment by spiffxp** [comment]
Also, FWIW, I have not yet been able to find anything that documents this "standard style", if there's a link I'd be super grateful

**Comment by johnSchnake** [comment]
My opinion on this is that my personal opinion doesn't matter much :-p TL;DR; I'm just on the side of thinking this is more bikeshedding than useful and (even though I might agree with the renaming) there are just as many people that would either disagree or not think it's worth the effort.

Elaboration:
I have spent way too much time on this sort of refactor including complaints about what my tooling automatically did (e.g. v1 alias instead of corev1), told to refactor things like this PR does, told it was too disruptive and to do smaller PRs, told there was a standard when there wasn't, etc.

At the end of the day I just settled on considering this a pet peeve that I may or may not fix up file by file when my PRs touch them, but have abandoned any intent to fix them wholesale for exactly the reasons you mentioned @spiffxp : to set the standard we :
 - (1) have to agree on it 
 - (2) have to do large, rebase ridden changes (or _lots_ of small ones) 
 - (3) this inevitably disrupts many devs and causes frustration 
 - (4) to enforce it we need tooling (and I also didnt see how to do something like that with goimports) 
 - (5) when I floated that idea of mandated goimports it was just shot down and there was no support for requiring something like that since goimports isnt as ubiquitous as gofmt.

**Review by cmluciano — COMMENTED** [review]
This looks good to me, but I'll hold off on appending the dash command until the feedback settles

**Comment by dims** [comment]
@johnSchnake i feel we have it backwards in youtr 1-5 analysis, as we found out in https://github.com/kubernetes/kubernetes/pull/76863#issuecomment-485265380 (and other input from folks who have tried it before) `goimports` does not really work well. hence the feeling that making a change that will just regress is not useful.

@spiffxp yes `it's not automatically enforced via a verify script` is the basic issue. If we can address that then we can do `(1)` that John outlined and then figure out when we could do it etc. (early in cycle or late in the cycle... each has its pros and cons, but we could come to a reasonable consensus)

**Comment by dims** [comment]
@cblecker reminded me of the slack thread - https://kubernetes.slack.com/archives/CHGFYJVAN/p1557526321126200

**Comment by spiffxp** [comment]
ref: https://github.com/kubernetes/kubernetes/pull/78836 for an idea of how to do enforcement

File: test/e2e/apimachinery/aggregator.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ import (
 	"strings"
 	"time"
 
-	apps "k8s.io/api/apps/v1"
+	appsv1 "k8s.io/api/apps/v1"
```
Comment (timothysc) [inline test/e2e/apimachinery/aggregator.go]: why?  unless there are conflicting imports, it's not really consistent in the codebase.

Comment (spiffxp) [reply test/e2e/apimachinery/aggregator.go]: See the description, this particular color for the bikeshed is to explicitly include the version, all the time.  I'm not trying to suggest we enforce this for the whole k/k codebase, just anything that could potentially be promoted to conformance

Comment (timothysc) [reply test/e2e/apimachinery/aggregator.go]: ¯\_(ツ)_/¯  - we tried this with goimports and we ended on ... "it's inconsistent and not enforced".

So where do we draw the lines and is it worth it.

**Comment by johnbelamaric** [comment]
Not sure I see the value here, Aaron. I don't think it adds a big burden to review the imports, and I don't think it causes a lot of confusion in the tests themselves for the names to be unversioned.

**Comment by dims** [comment]
So i have scripts https://github.com/kubernetes/kubernetes/pull/78836 that can do the update and verify for a set of alias(es) picked from a file.

**Comment by spiffxp** [comment]
So, we discussed giving this a try in k8s-code-organization. No enforcement, but try giving it a nudge every now and then in the test dirs and see if we hate it.  Given the way most e2e tests are authored via copy-paste it seems like it wouldn't drift much.

I used https://github.com/kubernetes/kubernetes/pull/78836 with some fixes and https://gist.github.com/spiffxp/e278d28755a3f64899aaed8f73c78ecf to generate a commit per api (and generalized from kubelet to component)

This is... a lot.  I feel like we've proven it can be done (and the script to do so _is_ really fast), but I still hear the "but why" loud and clear

**Comment by spiffxp** [comment]
rebased ontop of https://github.com/kubernetes/kubernetes/pull/78836

**Comment by spiffxp** [comment]
rebased now that https://github.com/kubernetes/kubernetes/pull/78836 merged

**Pull request merged**